workers/unity/Packages/com.improbable.gdk.tools/.CodeGenerator/GdkCodeGenerator/Templates/UnityComponentDataGenerator.tt

```diff
@@ -35,7 +107,7 @@ namespace <#= qualifiedNamespace #>
                 get => <#= fieldDetails.CamelCaseName #>;
                 set
                 {
-                    DirtyBit = true;
+                    MarkDataDirty(<#= i #>);
```

Comment:
Observation (no change needed) - this will be unstable with reordering two components in schema, in such a way that the schema doesn't change (field ids are the same).

I don't think this is a problem or something that we can do anything about in this PR. But might be something we need to be aware of if we are able to do something with serialising components.

Reply:
We could let `IsDataDirty()` and `MarkDataDirty()` use the schema field number of a property. In that case, we would scan for all field numbers and potentially create more byte bitmasks than necessary in case the field numbers have gaps but it would make the generated code more invariant.